File: VB/DXExample.Module/CustomUser.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace DXExample.Module
	<DefaultClassOptions> _
	Public Class CustomUser
		Inherits User
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		<Association("Users-Resources")> _
		Public ReadOnly Property Resources() As XPCollection(Of UserResource)
			Get
				Return GetCollection(Of UserResource)("Resources")
			End Get
		End Property
	End Class

End Namespace

File: CS/DXExample.Module/CustomUser.cs

using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace DXExample.Module {
    [DefaultClassOptions]
    public class CustomUser : User {
        public CustomUser(Session session) : base(session) { }
        [Association("Users-Resources")]
        public XPCollection<UserResource> Resources {
            get { return GetCollection<UserResource>("Resources"); }
        }
    }

}
